### PR TITLE
remove woocommerce_form_field radio type input label content HTML esc…

### LINKED ISSUE #26865: woocommerce_form_field function doesn't work as expected
Hi all. 
After a new Releas, when I use a value with html tags to radio oprions as parameters of woocommerce_form_field args, then the input value that I get in the browser is is not rendered html, because the added html elements are showing with html entities characters

### PR DESCRIPTION
**Pull request by anttiruonakoski**
…aping. fixes #26865

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Label content for input elements of radio type will be no longer HTML escaped when inputs are outputted by woocommerce_form_field(). This enables for example to style label content with any tags as span. Currently, styling is not possible, because output is forced as plain text. 

Closes #26865 .

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: includes/wc-template-functions.php [hunk 1]
```diff
@@ -2846,7 +2846,7 @@ function woocommerce_form_field( $key, $args, $value = null ) {
 				if ( ! empty( $args['options'] ) ) {
 					foreach ( $args['options'] as $option_key => $option_text ) {
 						$field .= '<input type="radio" class="input-radio ' . esc_attr( implode( ' ', $args['input_class'] ) ) . '" value="' . esc_attr( $option_key ) . '" name="' . esc_attr( $key ) . '" ' . implode( ' ', $custom_attributes ) . ' id="' . esc_attr( $args['id'] ) . '_' . esc_attr( $option_key ) . '"' . checked( $value, $option_key, false ) . ' />';
-						$field .= '<label for="' . esc_attr( $args['id'] ) . '_' . esc_attr( $option_key ) . '" class="radio ' . implode( ' ', $args['label_class'] ) . '">' . esc_html( $option_text ) . '</label>';
+						$field .= '<label for="' . esc_attr( $args['id'] ) . '_' . esc_attr( $option_key ) . '" class="radio ' . implode( ' ', $args['label_class'] ) . '">' . $option_text . '</label>';
```
Comment (claudiosanches) [inline includes/wc-template-functions.php:2849]: We still need to keep some escaping, so probably better to use `wp_kses_post()` that will allow some HTML.

```suggestion
						$field .= '<label for="' . esc_attr( $args['id'] ) . '_' . esc_attr( $option_key ) . '" class="radio ' . implode( ' ', $args['label_class'] ) . '">' . wp_kses_post( $option_text ) . '</label>';
```

**Review by claudiosanches — CHANGES_REQUESTED** [review]

**Comment by jorgeatorres** [comment]
Hi @anttiruonakoski,

Thanks for your contribution and apologies for the delay in getting back to you.

Overall this looks good but there are some conflicts that need to be resolved. Would you please rebase this branch to fix the conflicts? Also, even though the goal here is sort of clear, it'd be great to have some testing instructions (a snippet of code) to be able to properly test.

I can see it's been a while since you last worked on this, so please let us know if you're no longer interested or if you have any questions.

Thanks again!

**Comment by peterfabian** [comment]
Hi @anttiruonakoski ! Please let us know if you still want to update the PR based on @jorgeatorres 's comment. If not, we can take over. Thanks! 

Side note for next reviewer: we should double check the set of HTML tags we want to allow here.

**Comment by kaushikasomaiya** [comment]
Can anyone confirm if the radio field type supports validation `required` in WooCommerce default validation?

**Comment by peterfabian** [comment]
@kaushikasomaiya Can you please elaborate? Looking at the mozilla docs, this is what they say about required + radio:

> In the case of a same named group of [radio](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio) buttons, if a single radio button in the group has the required attribute, a radio button in that group must be checked, although it doesn't have to be the one with the attribute is applied. So to improve code maintenance, it is recommended to either include the required attribute in every same-named radio button in the group, or else in none.

As I understand it, the validation happens in the browser, not within WooCommerce?

**Comment by kaushikasomaiya** [comment]
> As I understand it, the validation happens in the browser, not within WooCommerce?

@peterfabian I will do a test and get back to you. WooCommerce doesn't use HTML5 browser validation, it has its own validation. Last time I did some tests with user's code [here](https://wordpress.org/support/topic/checkout-woocommerce-invalid-class-for-radio/), I wasn't able to get the radio field validated.

But I will do some more tests to confirm it. 

`required` is an array key in `woocommerce_form_field()` function.

**Comment by Konamiman** [comment]
Assigning a reviewer from Proton so that we can continue taking a look at this.

**Comment by samueljseay** [comment]
We are closing this pull request, as the last activity has been quite some time ago and we have not received a response from you. If you are still interested, please open another PR based on a recent version of the trunk branch.

cc @vedanshujain @Konamiman

**Comment by vedanshujain** [comment]
This will need a new PR to apply the changes against trunk since current changes are pre-monorepo. Re-opening anyway to track that work.

**Comment by jonathansadowski** [comment]
@vedanshujain,

Thanks for the follow-up. I'm going to close this, as we're trying to reduce the number of open PRs that aren't immediately actionable. Please create an issue or a new PR if you'd like to track this as opposed to re-opening this PR.

Context: pdqwkG-JU-p2

**Pull request closed without merging**